File: backend/app/schemas/audit.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel


class AuditLogItem(BaseModel):
    id: int
    created_at: datetime | None = None
    actor: str | None = ""
    action: str | None = ""
    target_type: str | None = ""
    target_id: str | None = ""
    target_name: str | None = ""
    server_name: str | None = ""
    task_id: str | None = ""
    status: str | None = ""
    message: str | None = ""
    detail_json: Any = None
    client_ip: str | None = ""

    class Config:
        from_attributes = True


class AuditLogPage(BaseModel):
    items: list[AuditLogItem]
    total: int
    page: int
    page_size: int

File: frontend/src/api/audit.ts
import { request } from './request'

export interface AuditLogItem {
  id: number
  created_at: string | null
  actor: string | null
  action: string | null
  target_type: string | null
  target_id: string | null
  target_name: string | null
  server_id: number | null
  server_name: string | null
  task_id: string | null
  status: string | null
  message: string | null
  detail_json: any
  client_ip: string | null
}

export interface AuditLogPage {
  items: AuditLogItem[]
  total: number
  page: number
  page_size: number
}

export interface AuditLogQuery {
  page?: number
  page_size?: number
  action?: string
  target_type?: string
  status?: string
  keyword?: string
  start_time?: string
  end_time?: string
  risk_only?: boolean
}

export function listAuditLogs(params: AuditLogQuery) {
  return request.get<AuditLogPage>('/audit-logs', { params })
}
